jerry-core/ecma/operations/ecma-function-object.c

```diff
@@ -1390,27 +1454,47 @@ ecma_op_function_call (ecma_object_t *func_obj_p, /**< Function object */
 
   ecma_value_t result;
 
-  if (JERRY_LIKELY (type == ECMA_OBJECT_TYPE_FUNCTION))
-  {
-    result = ecma_op_function_call_simple (func_obj_p, this_arg_value, arguments_list_p, arguments_list_len);
-  }
-  else if (type == ECMA_OBJECT_TYPE_BUILT_IN_FUNCTION)
-  {
-    result = ecma_op_function_call_native_built_in (func_obj_p, this_arg_value, arguments_list_p, arguments_list_len);
-  }
-  else if (type == ECMA_OBJECT_TYPE_NATIVE_FUNCTION)
+  switch (ecma_get_object_type (func_obj_p))
```

Comment:
The original code was in likelyhood order. Is this faster?

Reply:
Based on the benchmark results below it is.